controllers/commentController.js

```diff
@@ -50,4 +51,99 @@ export default class CommentController {
       return res.status(500).send({ message: error, status: 500 });
     }
   }
+
+
+  /**
+   * @description It allows a user to like a specific comment.
+   *
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - Returns the like created.
+   */
+
+  static async like(req, res) {
+    const { commentId } = req.params;
+    try {
+      const comments = await Comment.findOne({ where: { id: commentId } });
+      if (!comments) {
+        throw new Error("You are liking a non-existing comment");
+      }
+    } catch (error) {
+      if (error.message == "You are liking a non-existing comment") {
+        return res
+          .status(400)
+          .json({ message: "You are liking a non-existing comment" });
+      }
+      return res.status(500).json({ error: error.message });
+    }
+
+    try {
+      const userId = req.user.id;
+      const likes = await commentlike.findAll({ where: { commentId, userId } });
+      if (likes.length == 0) {
+        const likeComment = await commentlike.create({ commentId, userId });
+        const updateCommentLikes = await Comment.increment(
+          {
+            like: 1
+          },
+          { where: { id: commentId } }
+        );
+        return res.status(CREATED).json({
+          message: "Comment liked",
+          likeComment,
+          updatedComment: updateCommentLikes[0][0]
+        });
+      }
+      const unlikeComment = await commentlike.destroy({
+        where: { commentId, userId }
+      });
+      const updateCommentLikes = await Comment.decrement(
+        {
+          like: 1
+        },
+        { where: { id: commentId } }
+      );
+      res.status(CREATED).json({
+        message: "Comment unliked",
+        unlikeComment,
+        updatedComment: updateCommentLikes[0][0]
+      });
+    } catch (error) {
+      res.status(BAD_REQUEST).json({ error: error.message });
+    }
+  }
+
+  /**
+   * @description It helps to get all users who liked a specific comment.
+   *
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - Returns the comment object.
+   */
+
+  static async getLikingUsers(req, res) {
+    const { commentId } = req.params;
+    try {
+      let comment = await Comment.findOne({
+        where: { id: commentId }
+      });
+      if (comment) {
+        comment = comment.toJSON();
+        comment.likedBy = [];
+        const likes = await commentlike.findAll({
+          where: { commentId: comment.id },
+          include: [{ model: User, as: "likedBy", attributes: ["username"] }]
+        });
+
+        likes.map(like => {
```

Comment:
Can you use queries with an association http://docs.sequelizejs.com/manual/tutorial/models-usage.html

Reply:
I think that way of using ```map``` wouldn't be scalable.